pulsar-broker/src/main/java/org/apache/pulsar/broker/web/ProcessHandlerFilter.java

```diff
@@ -27,24 +27,20 @@
 import javax.servlet.ServletResponse;
 import javax.ws.rs.core.MediaType;
 import org.apache.commons.lang3.StringUtils;
-import org.apache.pulsar.broker.PulsarService;
 import org.apache.pulsar.broker.intercept.BrokerInterceptor;
 
 public class ProcessHandlerFilter implements Filter {
 
     private final BrokerInterceptor interceptor;
-    private final boolean interceptorEnabled;
 
-    public ProcessHandlerFilter(PulsarService pulsar) {
-        this.interceptor = pulsar.getBrokerInterceptor();
-        this.interceptorEnabled = !pulsar.getConfig().getBrokerInterceptors().isEmpty();
+    public ProcessHandlerFilter(BrokerInterceptor brokerInterceptor) {
```

Comment:
It is worth asserting that `interceptor` is not `null`?

Reply:
Thanks. I used `Objects.requireNonNull`.